### ISSUE TITLE
[LifetimeSafety] False negative on assignment from Owner to Pointer

### ISSUE THREAD

**Issue by NeKon69** [llvm/llvm-project#198605]
The analysis misses this GSL Pointer assignment case:
```cpp
struct [[gsl::Owner]] Owner {
  Owner();
};
struct [[gsl::Pointer]] View {
  View();
  View &operator=(const Owner &o [[clang::lifetimebound]]);
  void use() const;
};
void f() {
  View v;
  {
    Owner o;
    v = o;  
  }        // o is destroyed here!
  v.use(); // no warning
}
```
Expected: warn that o does not live long enough.

https://godbolt.org/z/EMT4WWrnc

**Comment by NeKon69** [comment]
@usx95 Misclick?

**Comment by usx95** [comment]
oops. Yes.
